controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -67,6 +67,31 @@ func (r *StorageAutoscalerReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// if more than one storage autoscaler is present in the cluster with same device class and same storage cluster
+	// return error
+	storageAutoScalerList := &ocsv1.StorageAutoScalerList{}
+	err = r.List(ctx, storageAutoScalerList,
+		client.InNamespace(request.Namespace),
+		client.MatchingFields{"spec.storageCluster.name": storageAutoScaler.Spec.StorageCluster.Name, "spec.deviceClass": storageAutoScaler.Spec.DeviceClass},
```

Comment:
Does this work? Our CustomerResources don't support field selection.

Reply:
Yes it didnt worked we need to ` indexed fields in the cache` maybe I can list all the crs and query it